src/core/caniuse.js

```diff
@@ -0,0 +1,212 @@
+/*
+Module: "core/caniuse"
+Adds a caniuse support table for a "feature" #1238
+
+`conf.caniuse = { feature: key }` =>
+  .. the table is added before Copyright
+Optional settings:
+  `conf.caniuse.browsers` list of caniuse supported browser names
+    to be shown in the table or "ALL"
+    default: ["chrome", "firefox", "safari", "edge"]
+  `conf.caniuse.versions` number of browser versions to show
+  `conf.caniuse.maxAge` (in ms) local response cache duration
+  `conf.caniuse.apiURL` URL from where to fetch stats.
+    use {FEATURE} as placeholder in URL to replace it by a feature name
+*/
+
+import { semverCompare } from "core/utils";
+import IDBCache from "core/idb-cache";
+import { pub } from "core/pubsubhub";
+import "deps/hyperhtml";
+import caniuseCss from "deps/text!core/css/caniuse.css";
+
+export const name = "core/caniuse";
+
+export async function run(conf) {
+  if (!conf.caniuse) {
+    return;
+  }
+  // Opportunistically insert the style into the head to reduce FOUC.
+  const codeStyle = document.createElement("style");
+  codeStyle.textContent = caniuseCss;
+  document.head.appendChild(codeStyle);
+
+  normalizeConf(conf);
+  const { caniuse } = conf;
+  const cache = await new IDBCache("respec-caniuse", ["caniuse"]);
+  if (caniuse.feature) {
+    canIUse(
+      caniuse.feature,
+      document.querySelector(".head dl").lastChild,
+      cache,
+      caniuse
+    );
+  }
+}
+
+/**
+ * MUTATES `conf.caniuse` object to hold normalized configurarion
+ * @param {Object} conf   configuration settings
+ */
+function normalizeConf(conf) {
+  const DEFAULTS = {
+    maxAge: 24 * 60 * 60 * 1000, // 24 hours (in ms)
+    browsers: ["chrome", "firefox", "safari", "edge"],
+    versions: 4,
+  };
+
+  if (typeof conf.caniuse === "string") {
+    conf.caniuse = { feature: conf.caniuse, ...DEFAULTS };
+    return;
+  }
+  if (Array.isArray(conf.caniuse.browsers)) {
+    conf.caniuse.browsers = conf.caniuse.browsers.map(b => b.toLowerCase());
+  } else if (conf.caniuse.browsers !== "ALL") {
+    conf.caniuse.browsers = DEFAULTS.browsers;
+  }
+  if (conf.caniuse.maxAge === undefined) conf.caniuse.maxAge = DEFAULTS.maxAge;
+  if (!conf.caniuse.versions) conf.caniuse.versions = DEFAULTS.versions;
+}
+
+/**
+ * main canIUse function
+ * @param  {String} key              which api to look for
+ * @param  {Node} refNode            adds table after refNode
+ * @param {IDBCache} cache           IDB cache
+ * @param {Object} conf              normalized respecConfig.caniuse
```

Comment:
The `conf` contains the `key`, so we could pass `conf` as the first argument?

Reply:
Yes that sounds valid now as we longer plan to support `<section data-caniuse>` (It was used earlier like this: https://github.com/w3c/respec/pull/1555/commits/bddc74fd62889bfde10d64099546b3834db8fa6e#diff-7ac308e51d1448eeee88084fe250ff44L45)